cpp/include/cudf/io/parquet.hpp

```diff
@@ -376,6 +376,48 @@ table_with_metadata read_parquet(
  * @{
  * @file
  */
+class table_input_metadata;
+
+class column_in_metadata {
+  friend table_input_metadata;
+  std::string _name = "";
+  thrust::optional<bool> _nullable;  // Input only. Can be output also but will be
+                                     // redundant with actual info in column_view
+  bool _list_column_is_map = false;  // Group only.
+  // yes
+  bool _use_int96_timestamp = false;  // Input, Primitive only.
+  // bool _output_as_binary = false;
+  thrust::optional<uint8_t> _decimal_precision;  // Primitive only.
+  std::vector<column_in_metadata> children;
+
+ public:
+  void set_name(std::string const& name) { _name = name; }
+  void set_nullability(bool nullable) { _nullable = nullable; }
+  void set_list_column_as_map() { _list_column_is_map = true; }
+  void enable_int96_timestamps(bool req) { _use_int96_timestamp = req; }
```

Comment:
```suggestion
  void set_int96_timestamps(bool req) { _use_int96_timestamp = req; }
```
Seems more congruent with the rest of the things.